modules/m_whois.c

```diff
@@ -373,10 +375,23 @@ single_whois(struct Client *source_p, struct Client *target_p, int operspy)
 					target_p->name, buf);
 		}
 
+		/* fire the doing_whois_show_idle hook to allow modules to tell us whether to show the idle time */
+		hook_data_client_approval hdata_showidle;
+
+		hdata_showidle.client = source_p;
+		hdata_showidle.target = target_p;
+		hdata_showidle.approved = 1;
+
+		call_hook(doing_whois_show_idle_hook, &hdata_showidle);
+
 		sendto_one_numeric(source_p, RPL_WHOISIDLE, form_str(RPL_WHOISIDLE),
-				   target_p->name,
-				   (long)(rb_current_time() - target_p->localClient->last),
-				   (unsigned long)target_p->localClient->firsttime);
+			   target_p->name,
+			   hdata_showidle.approved ? (long)(rb_current_time() - target_p->localClient->last) : 0,
+			   (unsigned long)target_p->localClient->firsttime);
+
+		if (hdata_showidle.approved == 2 || hdata_showidle.approved == 0)
+			/* if the target has hidden their idle time, notify the source */
+			sendto_one_numeric(source_p, RPL_WHOISTEXT, "%s :is using a private idle time", target_p->name);
```

Comment:
This should use the form_str:

```suggestion
			sendto_one_numeric(source_p, RPL_WHOISTEXT, form_str(RPL_WHOISTEXT), target_p->name, "is using a private idle time");
```